src/layout.js

```diff
@@ -224,8 +224,8 @@ export function getNaturalDimensions(tagName) {
     temp.style.visibility = 'hidden';
     document.body.appendChild(temp);
     naturalDimensions_[tagName] = {
-      width: temp./*OK*/offsetWidth || 1,
-      height: temp./*OK*/offsetHeight || 1
+      width: (temp./*OK*/offsetWidth || 1) + 'px',
```

Comment:
You mentioned that for amp-audio would be nice to compute the height but set the width to auto. So then, its entry would probably become {width: 'auto', height: null}, and the condition in l. 218 would need to check for width or height missing?
Just figured I'd mention this, could make that a TODO somewhere on this function or naturalDimensions_.